client/js/commands/join.js

```diff
@@ -8,13 +8,42 @@ exports.input = function(args) {
 	const {vueApp} = require("../vue");
 
 	if (args.length > 0) {
-		const channel = args[0];
+		let channels = args[0];
 
-		if (channel.length > 0) {
-			const chan = utils.findCurrentNetworkChan(channel);
+		if (channels.length > 0) {
+			const chanTypes = vueApp.activeChannel.network.serverOptions.CHANTYPES;
+			const channelList = args[0].split(",");
+
+			if (chanTypes && chanTypes.length > 0) {
+				for (let c = 0; c < channelList.length; c++) {
+					let hasPrefix = false;
+
+					for (const prefix of chanTypes) {
+						if (channelList[c].startsWith(prefix)) {
```

Comment:
What about `chanTypes.includes(channelList[c][0])`? That would remove the loop.

We do the same here: https://github.com/thelounge/thelounge/blob/b8948856f3db0c0a693d9acdfa13c8ce3fd1160b/src/plugins/inputs/msg.js#L35-L41